Fixing C language information in general

TheSoftwareGuy opened the pull request:
- Fixed error in block comments
- Began refactoring the Data Types Header
- Renamed the Structures header to be something more accurate
- fixed a mistake in the for loop syntax
- fixed mistakes in the switch examples
- Renamed User Interface to File I/O, as it is a more accurate description

TheSoftwareGuy committed:
Fixed block comments for both C and C++

TheSoftwareGuy committed:
Did some cleaning of C.markdown, added information and corrections

TheSoftwareGuy committed:
Began refactoring c.markdown
refactoring away from the entries being the type names
making the entries basic variables, and structs. more to come.

TheSoftwareGuy commented:
Hey, I am pretty new with the whole github stuff. I actually thought this would go to just my github fork, but it turns out I was wrong. Doesn't matter though this stuff is more correct, and you can merge it as long as the syntax I used was actually correct.

TheSoftwareGuy commented:
These changes are based on what I was saying on reddit:

http://www.reddit.com/r/programming/comments/2mtiez/tldrcode_a_website_for_quick_code_reference/cm7mdet

CCoffie commented:
Sorry I'm actually just heading to bed right now. Everything looks good. I just need to test the changes. I'll test them the first thing in morning. Thanks for helping out.

TheSoftwareGuy committed:
Combined if, else, and else if into one subheading

TheSoftwareGuy commented:
No problem. Yeah I am heading to bed soon too. I'll make more changes for the morning

TheSoftwareGuy committed:
Finished Data Types Heading & added to flow control
added pointer and array types to data types heading
added a type for functions under flow control

TheSoftwareGuy committed:
added string type to data types header

TheSoftwareGuy committed:
added fopen to the File I/O heading.

CCoffie commented:
Thanks again for the help. All your changes should be live now.
